tests/test_evaluation/test_split_evaluation.py

```diff
@@ -89,3 +99,206 @@ def test_no_missing_splits(model, tasks, tmp_path):
+    assert len(last_evaluated_splits["MockMultilingualRetrievalTask"]) == 1
+    assert last_evaluated_splits["MockMultilingualRetrievalTask"] == ["test"]
+    assert sorted(results[0].languages) == ["eng", "fra"]
+    assert results[0].scores.keys() == {"test"}
+    assert len(results[0].scores["test"]) == 2
+
+
+def test_no_missing_languages(model, multilingual_tasks, tmp_path):
+    evaluation = MTEB(tasks=multilingual_tasks)
+    results = evaluation.run(
+        model,
+        eval_splits=["test"],
+        output_folder=str(tmp_path / "no_missing_lang_test"),
+        verbosity=2,
+        eval_langs=["eng", "fra"],
+    )
+    last_evaluated_splits = evaluation.get_last_evaluated_splits()
+    assert "MockMultilingualRetrievalTask" in last_evaluated_splits
+    assert len(last_evaluated_splits["MockMultilingualRetrievalTask"]) == 1
+    assert results[0].scores.keys() == {"test"}
+    assert len(results[0].scores["test"]) == 2
+    assert sorted(results[0].languages) == ["eng", "fra"]
+
+    evaluation = MTEB(tasks=multilingual_tasks)
+    results = evaluation.run(
+        model,
+        eval_splits=["test"],
+        output_folder=str(tmp_path / "no_missing_lang_test"),
+        verbosity=2,
+        eval_langs=["eng", "fra"],
+        overwrite_results=True,
+    )
+    last_evaluated_splits = evaluation.get_last_evaluated_splits()
+    assert "MockMultilingualRetrievalTask" in last_evaluated_splits
+    assert len(last_evaluated_splits["MockMultilingualRetrievalTask"]) == 0
+    assert results[0].scores.keys() == {"test"}
+    assert len(results[0].scores["test"]) == 2
+    assert sorted(results[0].languages) == ["eng", "fra"]
+
+
+def test_partial_languages(model, multilingual_tasks, tmp_path):
+    evaluation = MTEB(tasks=multilingual_tasks)
+    results = evaluation.run(
+        model,
+        eval_splits=["test"],
+        output_folder=str(tmp_path / "partial_lang_test"),
+        verbosity=2,
+        eval_langs=["fra"],
+    )
+    last_evaluated_splits = evaluation.get_last_evaluated_splits()
+    assert len(last_evaluated_splits["MockMultilingualRetrievalTask"]) == 1
+    assert last_evaluated_splits["MockMultilingualRetrievalTask"] == ["test"]
+    assert results[0].languages == ["fra"]
+    assert results[0].scores.keys() == {"test"}
+
+    results = evaluation.run(
+        model,
+        eval_splits=["test"],
+        output_folder=str(tmp_path / "partial_lang_test"),
+        verbosity=2,
+        eval_langs=["fra", "eng"],
+        overwrite_results=True,
+    )
+    last_evaluated_splits = evaluation.get_last_evaluated_splits()
+    assert len(last_evaluated_splits["MockMultilingualRetrievalTask"]) == 1
+    assert last_evaluated_splits["MockMultilingualRetrievalTask"] == ["test"]
+    assert sorted(results[0].languages) == ["eng", "fra"]
+    assert results[0].scores.keys() == {"test"}
+    assert len(results[0].scores["test"]) == 2
+
+
+def test_multilingual_multiple_splits_partial_langs_partial_splits(
```

Comment:
I was confused by the naming, as all langs are run, but not all splits are.
```suggestion
def test_multilingual_one_missing_split_no_missing_lang(
```